go/nbs/store.go

```diff
@@ -129,15 +149,70 @@ func (nbs *NomsBlockStore) Put(c chunks.Chunk) {
 	nbs.stats.PutLatency.SampleTime(dur)
 }
 
+type byteTableReader struct {
+	tableReader
+	h    addr
+	data []byte
+}
+
+func (br byteTableReader) ReadAt(p []byte, off int64) (n int, err error) {
+	d.PanicIfTrue(off+int64(len(p)) > int64(len(br.data)))
+	copy(p, br.data[off:off+int64(len(p))])
+	return len(p), nil
+}
+
+func (br byteTableReader) hash() addr {
+	return br.h
+}
+
+// MUST be called already holding the store lock. Makes durable the existing
+// memtable. Will block if there are already |nbs.persistLimit| concurrent
+// persist operations underway until one has completed.
+func (nbs *NomsBlockStore) persistMemTable() {
```

Comment:
Every time I find myself writing a method like this that relies on someone reading the comment and doing the right thing with locking, I die a little on the inside because I know that someone will hold it wrong and we'll have a horrible, insidious bug that takes us days to track down.